src/createAnimatedComponent/createAnimatedComponent.tsx

```diff
@@ -291,6 +293,29 @@ export function createAnimatedComponent(
     _updateNativeEvents(
       prevProps: AnimatedComponentProps<InitialComponentProps>
     ) {
+      // If the event view tag changes, we need to completely re-mount all events
+      const computedEventTag = this._getEventViewTag();
+      if (this._eventViewTag !== computedEventTag) {
+        // Remove all bindings from previous props that ran on the old viewTag
+        for (const key in prevProps) {
+          const prevProp = prevProps[key];
+          if (
+            has('workletEventHandler', prevProp) &&
+            prevProp.workletEventHandler instanceof WorkletEventHandler
+          ) {
+            prevProp.workletEventHandler.unregisterFromEvents(
+              this._eventViewTag
+            );
+          }
+        }
```

Comment:
It seems like a code duplication with `_detachNativeEvents`

Reply:
@piaskowyk `_detachNativeEvents` works on current props. This code needs to remove previous ones

Reply:
+1